xml_converter/src/attribute/festival_filter_gen.cpp

```diff
@@ -59,27 +59,34 @@ FestivalFilter parse_festival_filter(rapidxml::xml_attribute<>* input, vector<XM
 }
 
 string festival_filter_to_xml_attribute(const std::string& attribute_name, const FestivalFilter* value) {
-    string output = "";
+    vector<string> flag_values;
     if (value->dragonbash == true) {
-        output = output + "dragonbash";
+        flag_values.push_back("dragonbash");
     }
     if (value->festival_of_the_four_winds == true) {
-        output = output + "festivalofthefourwinds";
+        flag_values.push_back("festivalofthefourwinds");
     }
     if (value->halloween == true) {
-        output = output + "halloween";
+        flag_values.push_back("halloween");
     }
     if (value->lunar_new_year == true) {
-        output = output + "lunarnewyear";
+        flag_values.push_back("lunarnewyear");
     }
     if (value->super_adventure_festival == true) {
-        output = output + "superadventurefestival";
+        flag_values.push_back("superadventurefestival");
     }
     if (value->wintersday == true) {
-        output = output + "wintersday";
+        flag_values.push_back("wintersday");
     }
     if (value->none == true) {
-        output = output + "none";
+        flag_values.push_back("none");
+    }
+    string output = "";
+    for (size_t i = 0; i < flag_values.size(); ++i) {
+        output += flag_values[i];
+        if (i < flag_values.size() - 1) {
+            output += ",";
+        }
```

Comment:
use the stringhelper join function instead here. It does exactly what you want and (will soon have) an optimization to preallocate the full size of the output string instead of reallocating each iteration of the loop.